track_oracle/core/kwiver_io_base.txx

```diff
@@ -30,10 +30,18 @@ typedef char (&no_io_type)[2];
 typedef char yes_io_type;
 
 struct proxy{ template <typename U> proxy(const U&); };
+# if defined(_MSC_VER)
+#  pragma warning( push )
+#  pragma warning( disable : 4172 ) /* returning address of local variable */
```

Comment:
Generally, of course, no, but this code is used as part of the template metaprogramming for I/O routing inside track_oracle and as such should never actually be generated, only used to steer the template instantiations[1]. Originally, the functions had no definitions, just declarations; but gcc5 complained, so an earlier patch added empty implementations. Then visual studio complained that the empty implementations returned nothing, so hence this patch.

[1] Caveat: Not claiming to be a TMP pro. Haven't checked that this code isn't generated. But I am sure it's never used for anything.